conda/models/match_spec.py

```diff
@@ -33,6 +33,17 @@
 
 class MatchSpecType(type):
     def __call__(cls, spec_arg=None, **kwargs):
+        try:
+            return cls._call(spec_arg, **kwargs)
+        except InvalidSpec as e:
+            msg = ""
+            if spec_arg:
+                msg += f"{spec_arg}"
+            if kwargs:
+                msg += " " + ", ".join(f"{k}={v}" for k, v in kwargs.items())
```

Comment:
Isn't this the same info as is added to the f-string below on line 77?

Reply:
It is, and kept like that for backwards compatibility, but this also adds that error message to any other exceptions raised during `MatchSpec` initialization (like the version problem reported in the issue).

Reply:
Ah I see